pkg/streamingpromql/operators/aggregations/topkbottomk/range_query_test.go

```diff
@@ -0,0 +1,322 @@
+				labels.FromStrings("pod", "2", "foo", "1", "group", "B"),
+				labels.FromStrings("pod", "1", "foo", "1", "env", "test", "group", "B"),
+				labels.FromStrings("pod", "2", "foo", "0", "env", "test", "group", "B"),
+			},
+		},
+
+		"grouping with 'without', multiple grouping labels in alternative order": {
+			inputSeries: []labels.Labels{
+				labels.FromStrings("pod", "1", "foo", "1", "env", "prod", "group", "A"),
+				labels.FromStrings("pod", "1", "foo", "0", "env", "prod", "group", "B"),
+				labels.FromStrings("pod", "2", "foo", "1", "env", "test", "group", "C"),
+				labels.FromStrings("pod", "2", "foo", "0", "env", "prod", "group", "B"),
+				labels.FromStrings("group", "D", "something-else", "yes"), // Test some series that have none of the grouping labels.
+				labels.FromStrings("group", "D"),
+				labels.FromStrings("pod", "1", "foo", "1", "env", "test", "group", "A"),
+				labels.FromStrings("pod", "1", "foo", "0", "env", "prod"), // Test some series that have some of the grouping labels.
+				labels.FromStrings("pod", "2", "foo", "1", "env", "test", "group", "A"),
+				labels.FromStrings("pod", "2", "foo", "0", "env", "prod", "group", "A"),
+				labels.FromStrings("pod", "1", "foo", "1", "env", "test", "group", "B"),
+				labels.FromStrings("pod", "1", "foo", "0", "group", "B"),
+				labels.FromStrings("pod", "2", "foo", "1", "env", "prod"),
+				labels.FromStrings("pod", "2", "foo", "0", "env", "prod", "group", "C"),
+				labels.FromStrings("pod", "2", "foo", "1", "group", "B"),
+				labels.FromStrings("pod", "2", "foo", "0", "env", "test", "group", "B"),
+			},
+			grouping: []string{"foo", "pod"},
+			without:  true,
+			expectedOutputSeriesOrder: []labels.Labels{
+				// Should sort so that the group that is completed first is returned first.
+				labels.FromStrings("pod", "2", "foo", "1", "env", "test", "group", "C"),
+				labels.FromStrings("pod", "1", "foo", "0", "env", "prod", "group", "B"),
+				labels.FromStrings("pod", "2", "foo", "0", "env", "prod", "group", "B"),
+				labels.FromStrings("group", "D", "something-else", "yes"),
+				labels.FromStrings("group", "D"),
+				labels.FromStrings("pod", "1", "foo", "1", "env", "test", "group", "A"),
+				labels.FromStrings("pod", "2", "foo", "1", "env", "test", "group", "A"),
+				labels.FromStrings("pod", "1", "foo", "1", "env", "prod", "group", "A"),
+				labels.FromStrings("pod", "2", "foo", "0", "env", "prod", "group", "A"),
+				labels.FromStrings("pod", "1", "foo", "0", "env", "prod"),
+				labels.FromStrings("pod", "2", "foo", "1", "env", "prod"),
+				labels.FromStrings("pod", "2", "foo", "0", "env", "prod", "group", "C"),
+				labels.FromStrings("pod", "1", "foo", "0", "group", "B"),
+				labels.FromStrings("pod", "2", "foo", "1", "group", "B"),
+				labels.FromStrings("pod", "1", "foo", "1", "env", "test", "group", "B"),
+				labels.FromStrings("pod", "2", "foo", "0", "env", "test", "group", "B"),
+			},
+		},
+	}
+
+	for name, testCase := range testCases {
```

Comment:
Can we not also use this to test the instant queries?

Reply:
The sorting behaviour for instant queries is quite different:

* in range queries, there is no defined sorting behaviour for `topk` and `bottomk`, so we sort series in the order that minimises the number we have to hold in memory
* in instant queries, we have to follow the sorting behaviour explained in https://github.com/prometheus/prometheus/pull/16009, and this is covered by the upstream tests with the exception of the grouping case, which I'll add something for now